### PR TITLE
support to access database by sql interface

### PR DESCRIPTION
**Pull request by tier-cap**
Function enhancement:
    Expose the DB's SQL interface.
    Decoupling part of the DB layer and workload layer

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -8,9 +8,9 @@ ifdef FDB_CHECK
 endif 
 
 ifeq ($(ROCKSDB_CHECK), 0)
-	TAGS += rocksdb
-    CGO_CXXFLAGS := "${CGO_CXXFLAGS} -std=c++11"
-    CGO_FLAGS += CGO_CXXFLAGS=$(CGO_CXXFLAGS)
+	# TAGS += rocksdb
```
Comment (hicqu) [inline Makefile:11]: Please revert it, it's only for environments without rocksdb libraries.

**Comment by hicqu** [comment]
It's also necessary to add methods
```go
type Workload interface {
    Init(db DB, load bool) error // create databases and tables in it if necessary
    Load(ctx context.Context, db DB, totalCount int64) error
}
```

And then for the core workload, create databases and tables in `Init` instead of `DB.Create`.

File: pkg/workload/core.go [hunk 2]
```diff
@@ -133,6 +133,16 @@ func createOperationGenerator(p *properties.Properties) *generator.Discrete {
 	return operationChooser
 }
 
+// Init implements the Workload Close interface.
+func (c *core) Init(db ycsb.DB) error {
+	return nil
+}
+
+// Load implements the Workload Close interface.
```
Comment (hicqu) [inline pkg/workload/core.go:141]: s/Close interface/Load interface

File: db/mysql/db.go [hunk 3]
```diff
@@ -43,7 +42,7 @@ const (
 type mysqlCreator struct {
 }
 
-type mysqlDB struct {
+type MysqlDB struct {
```
Comment (hicqu) [inline db/mysql/db.go:45]: It's not necessary to expose it.

**Comment by hicqu** [comment]
rest LGTM.

**Review by hicqu — APPROVED** [review]

File: pkg/workload/core.go [hunk 4]
```diff
@@ -133,6 +133,42 @@ func createOperationGenerator(p *properties.Properties) *generator.Discrete {
 	return operationChooser
 }
 
+// Init --create all schemas for ycsb workload
+func (c *core) Init(db ycsb.DB) error {
+	// need to redesign the relation btw worklaod and db interface later.
```
Comment (cosven) [inline pkg/workload/core.go:138]: typo: workload

File: pkg/workload/core.go [hunk 5]
```diff
@@ -133,6 +133,42 @@ func createOperationGenerator(p *properties.Properties) *generator.Discrete {
...
+		for i := int64(0); i < fieldCount; i++ {
+			buf.WriteString(fmt.Sprintf(", FIELD%d VARCHAR(%d)", i, fieldLength))
+		}
+
+		buf.WriteString(");")
+
+		_, err := sqlDB.Exec(buf.String())
+		return err
+	}
+	return nil
```
Comment (cosven) [inline pkg/workload/core.go:164]: Is the init step designed only for sql db?
Maybe the Init step should return an error when sqlDB is nil?

Comment (tier-cap) [reply pkg/workload/core.go:164]: 1. this init step currently only supports to init ycsb schema on mysql db, other dbs' are still created when DB.Create, so, currently this function return nil.
2. for the apis between workload and db interface, decoupling need to be done. and this PR is only for mysql DB, the ToSqlDB() api is also not suitable here. all these jobs will be done in next PR.

**Review by cosven — APPROVED** [review]

**Pull request merged**